src/sentry/coreapi.py

```diff
@@ -57,15 +51,6 @@ class APIForbidden(APIError):
     http_status = 403
 
 
-class APIRateLimited(APIError):
-    http_status = 429
-    msg = "Creation of this event was denied due to rate limiting"
-    name = "rate_limit"
-
-    def __init__(self, retry_after=None):
-        self.retry_after = retry_after
-
-
```

Comment:
Can't comment at the exact location, thus doing it here: `ClientApiHelper` is now only used for `insert_data_to_database`. Can we strip it down, or rather make it a plain function instead? The only cases where it is being used is:
- [Monitors](https://github.com/getsentry/sentry/blob/9d9545957138eea6a003f587d5d2441052e48a4e/src/sentry/models/monitor.py#L204-L205)
- [Reprocessing](https://github.com/getsentry/sentry/blob/95767d455b8004ec4b4c5026d84b64b6348e6d37/src/sentry/tasks/reprocessing.py#L29-L31)
- And tests (2x)

It would make sense to follow up with that, and see if we can post these to the ingest consumers rather than spawn the job. That said, we shouldn't do that in the same PR.